### ISSUE TITLE
client_id is missing from request body

### ISSUE THREAD

**Issue by yongheyan** [AzureAD/microsoft-authentication-library-for-java#472]
I have the following code:
```java
public class AccessTokenManager {
    private PrivateKey privateKey;
    private X509Certificate publicCert;

    private static final Log log = LogFactory.getLog(AccessTokenManager.class);

    public AccessTokenManager(PrivateKey privateKey, X509Certificate publicCert) {
        this.privateKey = privateKey;
        this.publicCert = publicCert;
    }

    public String retrieveAccessToken(String adfsEndpoint, String clientId, String resourceId) throws Exception {
        log.info("Retrieving Authentication token");

        PublicClientApplication a = PublicClientApplication.builder(clientId)
                .authority(adfsEndpoint)
                .build();

        IClientCredential credential = ClientCredentialFactory.createFromCertificate(privateKey, publicCert);
        ConfidentialClientApplication app = ConfidentialClientApplication
                .builder(clientId, credential)
                .authority(adfsEndpoint)
                .sendX5c(false)
                .build();
        ClientCredentialParameters parameters = ClientCredentialParameters
                .builder(Set.of(String.format("%s/openid", resourceId)))
                .build();

        IAuthenticationResult result = app.acquireToken(parameters).join();

        log.info("Retrieved Authentication token");
        return result.accessToken();
    }
}
```

I tried to acquire token like these:
```java
        AccessTokenManager m = new AccessTokenManager(privateKey, cert);
        String token = m.retrieveAccessToken("https://my.server.com/adfs", "my-client-id","http://myResource" );
```
The server's response is something like "client_id is missing".
I checked the actual httpRequest and the body is like this:
```
client_info=1&client_assertion_type=urn%3Aietf%3Aparams%3Aoauth%3Aclient-assertion-type%3Ajwt-bearer&grant_type=client_credentials&scope=http%3A%2F%2FmyResource%2Fopenid+openid+profile+offline_access&
client_assertion=eyJ4NXQ<---removed--->vxTwXfw
```
The client_id could not be found in the request body. However, according to Microsoft documentation from here: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow, the request body should contains client_id in the body like this:
```
  POST /{tenant}/oauth2/v2.0/token HTTP/1.1           //Line breaks for clarity
  Host: login.microsoftonline.com
  Content-Type: application/x-www-form-urlencoded
  
  client_id=535fb089-9ff3-47b6-9bfb-4f1264799865
  &scope=https%3A%2F%2Fgraph.microsoft.com%2F.default
  &client_secret=sampleCredentia1s
  &grant_type=client_credentials
```

I am wondering why client_id is missing from the request body?

**Comment by yongheyan** [comment]
@siddhijain The version is 1.11.1.

**Comment by siddhijain** [comment]
@yongheyan Please use the latest version 1.11.2. Let me know if that resolves the issue.

**Comment by siddhijain** [comment]
Hello @yongheyan. You're right about client_id missing from the request body and we will work on fixing this in one of the next releases. However, I was still able to get an access token using the code above. Can you please check?

**Comment by yongheyan** [comment]
@siddhijain: Our ADFS server still responses with an error message that says: client_id is missing.

**Comment by yongheyan** [comment]
Any update on this issue?

**Comment by siddhijain** [comment]
Hello, we are working on a fix and will update the thread when it is available.

**Comment by Avery-Dunn** [comment]
This issue was addressed in https://github.com/AzureAD/microsoft-authentication-library-for-java/pull/489, and should be fixed as of the recent [1.12.0 release](https://search.maven.org/artifact/com.microsoft.azure/msal4j/1.12.0/jar).

If the issue is still unresolved or if there are any related issues feel free to re-open this thread or open a new one.
